src/xo-app/migrate-vm/index.js

```diff
@@ -0,0 +1,245 @@
+import _ from 'messages'
+import forEach from 'lodash/forEach'
+import Icon from 'icon'
+import find from 'lodash/find'
+import map from 'lodash/map'
+import React, { Component } from 'react'
+import SingleLineRow from 'single-line-row'
+import { Col } from 'grid'
+import {
+  SelectHost,
+  SelectNetwork,
+  SelectSr
+} from 'select-objects'
+import {
+  connectStore,
+  mapPlus
+} from 'utils'
+import {
+  createGetObjectsOfType,
+  createPicker,
+  createSelector
+} from 'selectors'
+
+import styles from './index.css'
+
+@connectStore(() => {
+  const getVm = (_, props) => props.vm
+
+  const getVbds = createGetObjectsOfType('VBD').pick(
+    (state, props) => getVm(state, props).$VBDs
+  ).sort()
+
+  const getVdis = createGetObjectsOfType('VDI').pick(
+    createSelector(
+      getVbds,
+      vbds => mapPlus(vbds, (vbd, push) => {
+        if (!vbd.is_cd_drive && vbd.VDI) {
+          push(vbd.VDI)
+        }
+      })
+    )
+  )
+
+  const getVifs = createGetObjectsOfType('VIF').pick(
+    createSelector(getVm, vm => vm.VIFs)
+  ).sort()
+
+  const getPifs = createGetObjectsOfType('PIF')
+  const getNetworks = createGetObjectsOfType('network')
+  const getSrs = createGetObjectsOfType('SR')
+  const getPools = createGetObjectsOfType('pool')
+
+  return (state, props) => ({
+    networks: getNetworks(state, props),
+    pifs: getPifs(state, props),
+    pools: getPools(state, props),
+    srs: getSrs(state, props),
+    vdis: getVdis(state, props),
+    vifs: getVifs(state, props)
+  })
+}, { withRef: true })
+export default class MigrateVmModalBody extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      mapVdisSrs: {},
+      mapVifsNetworks: {}
+    }
+
+    this._getHostPredicate = createSelector(
+      () => this.props.vm,
+      ({ $container }) => host => host.id !== $container
+    )
+
+    this._getSrPredicate = createSelector(
+      () => this.state.host,
+      host => (host
+        ? sr => sr.content_type !== 'iso' && (sr.$container === host.id || sr.$container === host.$pool)
+        : () => false)
+    )
+
+    this._getNetworkPredicate = createSelector(
+      createPicker(
+        () => this.props.pifs,
+        createSelector(
```

Comment:
No need to use `createSelector()` for that, because the result function is fast and does not create new values (array, object, function, …).